File: Leetcode/1 - Two Sum/solution.js
var twoSum = function(nums, target) {
    const numsIdx = new Map();
    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i];
        if (numsIdx.has(complement)) {
            return [numsIdx.get(complement), i];
        }
        numsIdx.set(nums[i], i);
    }
};

File: Leetcode/1 - Two Sum/solution.cpp
vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int, int> numsIdx;
    for (int i = 0; i < nums.size(); i++) {
        const int complement = target - nums[i];
        if (numsIdx.find(complement) != numsIdx.end()) {
            return {numsIdx[complement], i};
        }
        numsIdx[nums[i]] = i;
    }
}